Fix mktemp invocation in ui makefile

on macOS:

```
# mktemp ./elm-XXXXXXXXXX.js
./elm-XXXXXXXXXX.js
~/s/g/p/a/u/app (master *%=)
# ls
CONTRIBUTING.md   README.md         elm-stuff         favicon.ico       script.js         tests
Makefile          elm-package.json  elm-XXXXXXXXXX.js index.html        src
```

mktemp isn't using the provided template.

@mxinden yeah, I could reproduce this on macOS

```sh
make build-all
cd /Users/w0rm/Work/Go/src/github.com/prometheus/alertmanager/ui/app && /Applications/Xcode.app/Contents/Developer/usr/bin/make script.js
>> building elm-env docker image
>> format front-end code
Processing multiple files...
>> building script.js
Success! Compiled 1 module.                                         
Successfully generated ./elm-XXXXXXXXXX.js
make: go-bindata: No such file or directory
make: *** [ui/bindata.go] Error 1
```

--- a/ui/app/Makefile
+++ b/ui/app/Makefile
@@ -1,7 +1,10 @@
 ELM_FILES := $(shell find src -iname *.elm)
 DOCKER_IMG :=elm-env
 DOCKER_CMD := docker run --rm -t -v $(PWD):/app -w /app $(DOCKER_IMG)
+# macOS requires mktemp template to be at the end of the filename.
 TEMPFILE := $(shell mktemp ./elm-XXXXXXXXXX)
+# --output flag for elm make must end in .js or .html.
+TEMPFILE_JS := "$(TEMPFILE).js"
 
 ifeq ($(NO_DOCKER), true)
   DOCKER_CMD=
@@ -28,9 +31,9 @@ dev-server:
 
 script.js: elm-env format $(ELM_FILES)
 	@echo ">> building script.js"
-	@$(DOCKER_CMD) elm make src/Main.elm --yes --output $(TEMPFILE)
-	@$(DOCKER_CMD)	uglifyjs $(TEMPFILE) --compress unused --mangle --output $(@)
-	@rm $(TEMPFILE)
+	@$(DOCKER_CMD) elm make src/Main.elm --yes --output $(TEMPFILE_JS)
+	@$(DOCKER_CMD)	uglifyjs $(TEMPFILE_JS) --compress unused --mangle --output $(@)
+	@rm $(TEMPFILE_JS) $(TEMPFILE)
 
 clean:
 	- @rm script.js

--- a/ui/app/Makefile
+++ b/ui/app/Makefile
@@ -33,7 +33,7 @@ script.js: elm-env format $(ELM_FILES)
 	@echo ">> building script.js"
 	@$(DOCKER_CMD) elm make src/Main.elm --yes --output $(TEMPFILE_JS)
 	@$(DOCKER_CMD)	uglifyjs $(TEMPFILE_JS) --compress unused --mangle --output $(@)
-	@rm $(TEMPFILE_JS) $(TEMPFILE)
+	@rm -rf $(TEMPFILE_JS) $(TEMPFILE)
 
 clean:
 	- @rm script.js

--- a/ui/app/Makefile
+++ b/ui/app/Makefile
@@ -1,7 +1,7 @@
 ELM_FILES := $(shell find src -iname *.elm)
 DOCKER_IMG :=elm-env
 DOCKER_CMD := docker run --rm -t -v $(PWD):/app -w /app $(DOCKER_IMG)
-TEMPFILE := $(shell mktemp ./elm-XXXXXXXXXX.js)
+TEMPFILE := $(shell mktemp ./elm-XXXXXXXXXX)
 
 ifeq ($(NO_DOCKER), true)
   DOCKER_CMD=

